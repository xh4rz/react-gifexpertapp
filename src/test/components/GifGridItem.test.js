import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGidItem/>', () => {
	const title = 'Un título';
	const url = 'https://localhost/algo.jpg';
	const wrapper = shallow(<GifGridItem title={title} url={url} />);

	test('debe de mostrar el componente correctamete', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de tener un párrafo con el title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});

	test('debe de tener la imagen igual al url y alt de los props', () => {
		const img = wrapper.find('img');
		// console.log(img.prop('src'));
		expect(img.prop('src')).toBe(url);
		expect(img.prop('alt')).toBe(title);
	});

	test('debe tene animate__fadeIn', () => {
		const div = wrapper.find('div');

		const className = div.prop('className');

		expect(className.includes('animate__fadeInDown')).toBe(true);
	});
});
