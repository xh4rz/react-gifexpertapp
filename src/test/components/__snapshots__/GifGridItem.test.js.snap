// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGidItem/> debe de mostrar el componente correctamete 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
